js/api/geolocation/index.js

```diff
@@ -0,0 +1,26 @@
+// Configuration for geolocation options, using gps, timeout and age
+const GEO_CONFIG = {
+  enableHighAccuracy: true, 
+  timeout: 10000, 
+  maximumAge: 1000,
+};
+
+// Returns the location if not timeout
+export const geolocation = () => {
+  return new Promise((resolve, reject) => {
+    navigator.geolocation.getCurrentPosition( 
+      (position) => { 
+        resolve(position); 
+      }, 
+      (error) => {
+        reject(error);
+      }, 
+      GEO_CONFIG 
+    ); 
+  });
+};
+
+// Watch for any changes in the location of the user
+export const geoWatch = (onSuccess, onReject) => {
+  return navigator.geolocation.watchPosition(onSuccess, onReject, GEO_CONFIG);
```

Comment:
`onSuccess` sounds like a one time event, isn't this callback called many times? If so it should be called `onChange` or `onUpdate` with `onError`.